pkg/identities/service.go

```diff
@@ -0,0 +1,101 @@
+package identities
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/monitoring"
+	kClient "github.com/ory/kratos-client-go"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	kratos kClient.IdentityApi
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+type IdentityData struct {
+	Identities []kClient.Identity
+	Error      *kClient.GenericError
+}
+
+func (s *Service) buildListRequest(ctx context.Context, page, size int64, credID string) kClient.IdentityApiListIdentitiesRequest {
+	return s.kratos.ListIdentities(ctx).Page(page).PerPage(size).CredentialsIdentifier(credID)
+}
+
+func (s *Service) parseError(r *http.Response) *kClient.GenericError {
+	gerr := kClient.NewGenericErrorWithDefaults()
+
+	defer r.Body.Close()
+	body, _ := io.ReadAll(r.Body)
+
+	if err := json.Unmarshal(body, gerr); err != nil {
+		gerr.SetMessage("unable to parse kratos error response")
+		gerr.SetCode(http.StatusInternalServerError)
+	}
+
+	return gerr
+}
+
+func (s *Service) ListIdentities(ctx context.Context, page, size int64, credID string) (*IdentityData, error) {
+	_, span := s.tracer.Start(ctx, "kratos.IdentityApi.ListIdentities")
```

Comment:
A bit curious if we could use the decorator pattern to do the tracing. I guess it would be simplified for each endpoint handler.

Reply:
we already have that, this is to spread the trace further down each function, can show you what i mean in a call

Reply:
gotcha~ let's find a time after the stand-up tomorrow. This is a good chance to learn golang 😁